overlord/managers_test.go

```diff
@@ -14539,12 +14589,56 @@ func (s *mgrsSuite) TestConnectionDurabilityDuringRefreshes(c *C) {
 		},
 	})
 
+	// because of a bug fixed in 2.58, the repository may or may not
+	// correctly reflect the connection state, but the test is also rigged
+	// to not execute auto-connect, which would restore the connection in
+	// the scenario when pending security profiles were not carried
+	ref, err := s.o.InterfaceManager().Repository().Connected("snap-with-snapd-control", "snapd-control")
+	if hasPendingSecurityProfiles {
+		c.Assert(err, IsNil)
+		// with pending security profiles the repo information is correctly restored
+		c.Assert(len(ref) > 0, Equals, true, Commentf("connection not restored to interfaces repo"))
+		// CanManageRefreshes must return true with the >= 2.58 fixes
+		c.Check(devicestate.CanManageRefreshes(st), Equals, true)
+	} else {
+		// but without the connection present in the state is not
+		// reflected by the repo
+		c.Assert(err, ErrorMatches, `snap "snap-with-snapd-control" has no .* "snapd-control"`)
+		c.Assert(ref, HasLen, 0)
+		// but returns false without the fix, yet we still made no
```

Comment:
is the yet and no right here?  above we have c.Assert(len(reqs) > 0, Equals, true) ?

Reply:
sorry, yes, that code was slightly later (after the checks that auto refresh wasn't done), but then I moved it around without updating the comment